Source/WebCore/rendering/svg/LegacyRenderSVGShape.cpp

```diff
@@ -253,11 +238,8 @@ void LegacyRenderSVGShape::strokeShape(GraphicsContext& context)
         return;
 
     GraphicsContextStateSaver stateSaver(context, false);
-    if (hasNonScalingStroke()) {
-        AffineTransform nonScalingTransform = nonScalingStrokeTransform();
-        if (!setupNonScalingStrokeContext(nonScalingTransform, stateSaver))
-            return;
-    }
+    if (hasNonScalingStroke() && !setupNonScalingStrokeContext(stateSaver))
```

Comment:
Confused by this comment. Are you saying I should rewrite:

```c++
if (hasNonScalingStroke() && !setupNonScalingStrokeContext(...))
    return;
```

to

```c++
if (hasNonScalingStroke()) {
    if (!setupNonScalingStrokeContext(...)
        return;
}
```

?

Reply:
Sorry for the confusion. I fixed the comment. I meant adding the following lines to `setupNonScalingStrokeContext()`:

```
bool LegacyRenderSVGShape::setupNonScalingStrokeContext(GraphicsContextStateSaver& stateSaver)
{
    if (hasNonScalingStroke())
        return true;
    AffineTransform nonScalingTransform = nonScalingStrokeTransform();
    ...
}

```

And instead of calling it like this:

```
    if (hasNonScalingStroke()) {
        AffineTransform nonScalingTransform = nonScalingStrokeTransform();
        if (!setupNonScalingStrokeContext(nonScalingTransform, stateSaver))
            return;
    }
```

It can be just

```
    if (!setupNonScalingStrokeContext(stateSaver))
        return;

```